kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiServer.scala

```diff
@@ -84,8 +86,15 @@ class KyuubiServer(name: String) extends Serverable(name) {
 
   override val backendService: AbstractBackendService = new KyuubiBackendService()
 
-  override val frontendServices: Seq[AbstractFrontendService] = Seq(
-    new KyuubiThriftBinaryFrontendService(this))
+  override lazy val frontendServices: Seq[AbstractFrontendService] =
+    conf.get(FRONTEND_PROTOCOLS).map(FrontendProtocols.withName).map {
+      case THRIFT_BINARY => new KyuubiThriftBinaryFrontendService(this)
+      case REST =>
+        warn("REST frontend protocol is experimental, API may changes in the future.")
+        new RestFrontendService(this)
+      case us =>
```

Comment:
what does the `us` mean? `unsupported`? `other` looks better?

Reply:
yea, `us` means `unsupported`, I'm fine with `other`